src/AssemblyGenerator.Types.cs

```diff
@@ -1,95 +1,146 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 using System.Reflection;
 using System.Reflection.Metadata;
 using System.Reflection.Metadata.Ecma335;
+using Lokad.ILPack.Metadata;
 
 namespace Lokad.ILPack
 {
     public partial class AssemblyGenerator
     {
-        private void CreateTypes(Type[] types)
+        private IEnumerable<Type> GetBaseTypes(Type type)
```

Comment:
I would suggest to add a method-level comment to explain the _why_. Not obvious, we missed that part the first time.

Reply:
Ok. I'll do it with next update.